_releases/2022-10-06-1.6.0-released.md

```diff
@@ -0,0 +1,73 @@
+
+Pre-built packages are available on [GitHub Releases](https://github.com/crystal-lang/crystal/releases/tag/1.6.0) and our official distribution channels.
+See [crystal-lang.org/install](https://crystal-lang.org/install/) for installation instructions.
+
+## Stats
+
+In this release we included [183 changes since the 1.5.1 release](https://github.com/crystal-lang/crystal/pulls?q=is%3Apr+milestone%3A1.6.0) by 26 contributors. We thank all the effort put into improving the language! ❤️
+
+Below we list the most remarkable changes in the language and stdlib.
+
+## ⚠️ Refactor in console methods
+
+By making some console methods available in Windows, a significant [refactor](https://github.com/crystal-lang/crystal/pull/12352) lead to deprecating three macro methods in `FileDescriptor`: `cooked_from_tc_mode!`, `noecho_from_tc_mode!`, and `raw_from_tc_mode!`. Additionally, methods `#noecho!` and `#raw!` now return `nil`.
+
+## ⚠️ Refactors in `File::Info`
+
+[Another refactor](https://github.com/crystal-lang/crystal/pull/12385) lead to an improvement in the `File` API, which was breaking the abstraction by returning an object intended to be internal. In practice, if for some reason you ended up with an object of type `Crystal::System::FileInfo`, it should now be `File::Info`.
+
+## ⚠️ Improvements in overload ordering
+
+[Union types are now considered first](https://github.com/crystal-lang/crystal/pull/12335) when searching for a suitable overload. Consider the following code:
+
+```crystal
+module Foo(T)
+end
+
+class Bar1
+  include Foo(Int32)
+end
+
+class Bar2
+  include Foo(Int32)
+end
+
+def foo(x : Foo(Int32))
+  'a'
+end
+
+def foo(x : Bar1 | Bar2)
+  true
+end
+
+foo(Bar1.new)
+```
+
+Before 1.6.0 it would print `'a'`, meaning the generic case was considered first. Now it correctly prints `true`.
+
+Several other [improvements](https://github.com/crystal-lang/crystal/pull/10711) where added to the overloading search algorithm, but only under a special flag `-Dpreview_overload_order`.
+
+## Improvements in the interpreter
+
+In this release [a number interpreter bugs](https://github.com/crystal-lang/crystal/pulls?q=is%3Apr+sort%3Aupdated-desc+milestone%3A1.6.0+label%3Atopic%3Acompiler%3Ainterpreter) got fixed. We are still working towards incorporating it in the major platforms; in the meantime you can build the compiler with interpreter support locally with `make interpreter=1`, see the [interpreter introduction post for more info](https://crystal-lang.org/2021/12/29/crystal-i.html).
+
+## Performance improvements
+
+[Little improvements](https://github.com/crystal-lang/crystal/pulls?q=is%3Apr+sort%3Aupdated-desc+milestone%3A1.6.0+label%3Aperformance) in the compiler and the stdlib went into this release. We don't have a benchmark for you, but users reported non-negligible improvements in compilation speed.
```

Comment:
It's worth mentioning that the improvements are not just about speed but also memory consumption.

Also can we do a quick benchmark to say something like "hello world program compiles 7% faster and uses 10% less memory with 1.6.0"?

Reply:
The increased IO buffer size might worth an explicit mention (https://github.com/crystal-lang/crystal/pull/12507).

Reply:
Sadly, I couldn't actually verify any noticeable improvement between 1.5.1 and 1.6.0. If anything changed, it's worsening 😢 
But that is probably attributed to other changes, of course (not sure what exactly, though). If at all, these performance improvements reduce the impact.
We need an benchmark suite to quantify the performance effects of compiler changes (https://github.com/crystal-lang/crystal/issues/5508).